cpp/include/cudf/reduction/detail/reduction.cuh

```diff
@@ -66,7 +66,7 @@ std::unique_ptr<scalar> reduce(InputIterator d_in,
 {
   auto const binary_op     = cudf::detail::cast_functor<OutputType>(op.get_binary_op());
   auto const initial_value = init.value_or(op.template get_identity<OutputType>());
-  auto dev_result          = cudf::detail::device_scalar<OutputType>{initial_value, stream, mr};
+  auto dev_result          = rmm::device_scalar<OutputType>{initial_value, stream, mr};
```

Comment:
This is just a logging note rather than a formal code review. 
This line is the only remaining loose end in this PR with `rmm::device_scalar`, and the plan is to address it in the upcoming draft PR #19608.